packages/atomic/cypress/integration/facets/category-facet.cypress.ts

```diff
@@ -1,617 +1,372 @@
-    .find('[part*="parent"]')
-    .its('length')
-    .should('eq', totalParents);
+  canadaHierarchy,
+  canadaHierarchyIndex,
+  CategoryFacetSelectors,
+  hierarchicalField,
+  defaultNumberOfValues,
+  togoHierarchy,
+} from './category-facet-selectors';
+import * as CategoryFacetAssertions from './category-facet-assertions';
+
+interface CategoryFacetSetupOptions {
+  field: string;
+  attributes: string;
+  executeFirstSearch: boolean;
+  withResultList: boolean;
 }
 
-function assertTotalChildrentMoreThan(value: number) {
-  cy.get(FacetAlias.facetShadow)
-    .find('[part="child"]')
-    .its('length')
-    .should('be.gt', value);
+// TODO: adapt new setup
+function setupCategoryFacet(options: Partial<CategoryFacetSetupOptions> = {}) {
+  const setupOptions: CategoryFacetSetupOptions = {
+    attributes: '',
+    executeFirstSearch: true,
+    field: hierarchicalField,
+    withResultList: false,
+    ...options,
+  };
+  setUpPage(
+    `<atomic-breadcrumb-manager></atomic-breadcrumb-manager>     
+    <atomic-category-facet field="${setupOptions.field}" label="Atlas" ${
+      setupOptions.attributes
+    }></atomic-category-facet>
+    ${
+      setupOptions.withResultList && '<atomic-result-list></atomic-result-list>'
+    }`,
+    setupOptions.executeFirstSearch
+  );
 }
 
-function clickOnCategoryFacetWithValue(value: string) {
-  cy.get(FacetAlias.facetUL)
-    .find(FacetSelectors.categoryFacetNextLevelButton)
-    .contains(value)
-    .click();
+function selectChildValueAt(index: number) {
+  CategoryFacetSelectors.childValue().eq(index).click();
 }
 
-describe('Category Facet with default setting', () => {
-  beforeEach(() => {
-    setupCategoryFacet(categoryFacetProp.field, categoryFacetProp.label);
-    createAliasShadow(categoryFacetProp.field, FacetSelectors.categoryFacet);
-    createAliasFacetUL(categoryFacetProp.field, FacetSelectors.categoryFacet);
-    cy.wait('@coveoSearch');
-    cy.wait('@coveoAnalytics');
-  });
-
-  it('Should load, pass accessibility test, have correct label, facetCount and no facetSearch', () => {
-    validateFacetComponentLoaded(
-      categoryFacetProp.label,
-      FacetSelectors.categoryFacet
-    );
-    cy.get(FacetAlias.facetShadow)
-      .find(FacetSelectors.facetSearchbox)
-      .should('not.exist');
-  });
-
-  it('Should contain ShowMore buttons but should not contain "All Categories" button', () => {
-    cy.get(FacetAlias.facetShadow)
-      .find(FacetSelectors.clearAllButton)
-      .should('not.exist');
-    cy.get(FacetAlias.facetShadow)
-      .find(FacetSelectors.showMoreButton)
-      .should('be.visible');
-  });
+describe('Category Facet Test Suites', () => {
+  describe('with default settings', () => {
+    function setupWithDefaultSettings() {
+      setupCategoryFacet();
+      cy.wait(RouteAlias.search);
+    }
+
+    describe('verify rendering', () => {
+      before(setupWithDefaultSettings);
+
+      CategoryFacetAssertions.assertAccessibility();
```

Comment:
Another approach (I'm not saying it would be better) would be to inverse the logic.

Here, we do "for a given setup, test/assert those N different features are okay". 

What if it was "for a given feature, create N different setup, and assert the given feature is okay".

eg:

```
for (const setup of [()=> setupWithDefaultSeting(), ()=> setupGoDeeperOneLevel(), ()=> setupSelectFirstParent(), .....]) {
  setup();
 CategoryFacetAssertions.assertFoo();
} 
```

Just food for thoughts.

Reply:
I still prefer the original way, as later we might need to reuse these small tests for E2E tests on index.html/or client similar environment.